dumper/app/src/main/resources/oracle-stats/native/db-objects-synonym-public.sql

```diff
@@ -0,0 +1,52 @@
+-- Copyright 2022-2024 Google LLC
+-- Copyright 2013-2021 CompilerWorks
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+SELECT
+  B.con_id "ConId",
+  'PUBLIC' "Owner",
+  'SYNONYM' "ObjectType",
+  B.editionable "Editionable",
+  -- This looks similar to filtering with WHERE and using count() instead of sum().
+  --
+  -- It is not similar. DB will see the LIKE inside a WHERE predicate and decide to
+  -- replace a HASH JOIN with NESTED LOOPS. The JOIN arguments have >10k rows each,
+  -- so performance-wise the nested loop would be terrible.
+  sum(
+    CASE WHEN B.object_name LIKE '/%' THEN 0
+    WHEN B.object_name LIKE 'BIN$%' THEN 0
+    ELSE 1 END
```

Comment:
Optional: You could potentially consider replacing `CASE` with the `TO_INTEGER` function, although I'm not sure it's  a simplification, because it introduces additional negations (`NOT`):

```
TO_INTEGER(
   B.object_name NOT LIKE '/%' AND B.object_name NOT LIKE 'BIN$%'
)
```

Reply:
From what I've read, boolean was only introduced to Oracle in release 23:

https://docs.oracle.com/en/database/oracle/oracle-database/23/sqlrf/Data-Types.html#GUID-285FFCA8-390D-4FA9-9A51-47B84EF5F83A

So I think we're stuck with CASE here